src/umpire/ResourceManager.cpp

```diff
@@ -214,7 +225,14 @@ ResourceManager::getAllocator(int id)
 
   auto allocator = m_allocators_by_id.find(id);
   if (allocator == m_allocators_by_id.end()) {
-    UMPIRE_ERROR("Allocator \"" << id << "\" not found.");
+    auto available_ids = getAllocatorIds();
+
+    const char* const delim = " ";
+
+    std::ostringstream id_list;
```

Comment:
I think it would perhaps be more useful to display both the ID and the Name.  Perhaps a helper function that both errors could use that will display ID=XXX, Name=SSS?

Reply:
You think? I was thinking that if someone was searching by id they wouldn't be concerned about the names...

Reply:
I was thinking that getting a list of IDs without names might not be helpful.  If they get just a list of IDs, how will they know which one is the right one?

At any rate, I don't think the additional information would hurt.

Reply:
I have a proposal... we could implement `<<` for Allocator, and have that print name and id?

Reply:
This is resolved in latest changes.